ikarus/finiteelements/mechanics/enhancedassumedstrains.hh

```diff
@@ -138,54 +143,93 @@ public:
     if (numberOfEASParameters != 0)
       easApplicabilityCheck();
     easVariant_.setEASType(numberOfEASParameters);
+    initializeState();
   }
 
+  /**
+   * \brief Gets the internal state variable alpha for the EAS element.
+   *
+   * \return Internal state variable (alpha).
+   */
+  const auto& alpha() const { return alpha_; }
+
 protected:
   void bindImpl() {
     assert(underlying().localView().bound());
     easVariant_.bind(underlying().localView().element().geometry());
+    initializeState();
+  }
+
+  /**
+   * \brief Updates the internal state variable alpha_ at the end of an iteration before the update of the displacements
+   * done by the non-linear solver. See \cite bieberLockingHourglassingNonlinear2024 for implementation details and
+   * further references.
+   *
+   * \param par The Requirement object.
+   * \param correction The correction in displacement (DeltaD) vector passed based on which the internal state variable
+   * alpha is to be updated.
+   */
+  void updateStateImpl(const Requirement& par, typename Traits::template VectorTypeConst<> correction) const {
+    using ScalarType = Traits::ctype;
+    easApplicabilityCheck();
+    if (isDisplacementBased())
+      return;
+    const auto& Rtilde      = calculateRtilde<ScalarType>(par);
+    const auto localdxBlock = Ikarus::FEHelper::localSolutionBlockVector<Traits, Eigen::VectorXd, double>(
+        correction, underlying().localView());
+    const auto localdx  = Dune::viewAsFlatEigenVector(localdxBlock);
+    decltype(auto) LMat = LMatFunc<ScalarType>();
+
+    auto correctAlpha = [&]<typename EAST>(const EAST& easFunction) {
+      constexpr int enhancedStrainSize = EAST::enhancedStrainSize;
+      Eigen::Matrix<double, enhancedStrainSize, enhancedStrainSize> D;
+      calculateDAndLMatrix(easFunction, par, D, LMat);
+      const decltype(alpha_) updateAlpha = D.inverse() * (Rtilde + (LMat * localdx).eval());
```

Comment:
If you wrap the whole RHS with `(...).eval()` the LHS can simply be `const auto`. This might be mopre readable,